compendium/DeclarativeServices/src/ComponentContextImpl.cpp

```diff
@@ -117,9 +117,23 @@ std::shared_ptr<void> ComponentContextImpl::LocateService(
   if (serviceMapItr != boundServicesCacheHandle->end()) {
     auto& serviceMaps = serviceMapItr->second;
     if (!serviceMaps.empty()) {
-      service = ExtractInterface(serviceMaps.at(0), type);
+      bool nullDueToFailure = false;
+      service = ExtractInterface(serviceMaps.at(0), type, nullDueToFailure);
+
+      // In the case of service being a nullptr, we throw because this implies
+      // that the construction or activation of the service failed.
+      if (!service && nullDueToFailure) {
+        std::string errMsg = "Service ";
+        errMsg += name;
+        errMsg += " with type ";
+        errMsg += type;
+        errMsg += " failed to construct or activate.";
+
+        throw ComponentException(errMsg);
+      }
```

Comment:
Add check for 1..1/n cardinality requirement to throw. Add new test which tries to get a service to a service with cardinality 0..1 (should get back nullptr and not throw)

Reply:
This should actually return a valid pointer AND not throw.